### PR TITLE
Feature  gt evaluation

### PR DESCRIPTION
**Pull request by anty-filidor**
Added ground truth evaluator

### CODE REVIEW

File: _configs/eval_ssm.yaml [hunk 1]
```diff
@@ -5,12 +5,38 @@ spreading_model:
     p_values: [0.25, 0.5, 0.75]
 
 networks: [
+  # "artificial_er",
```
Comment (mateuszStolarski) [inline _configs/eval_ssm.yaml:8]: (1) I suggest to uncomment all test networks that are here
https://github.com/network-science-lab/infmax-trainer-icm-mln/blob/master/_configs/ssnet.yaml

File: src/evaluators/gt_ss.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+class GroundTruth:
+
+    def __init__(self, nb_seeds: int, average_protocol: bool, average_p_value: bool) -> None:
+        self.nb_seeds = nb_seeds
+        self.average_protocol = average_protocol
+        self.average_p_value = average_p_value
+    
+    @staticmethod
+    def get_top_k(sp_raw: pd.DataFrame, protocol: str | None, p: float | None, budget: int) -> list[str]:
```
Comment (mateuszStolarski) [inline src/evaluators/gt_ss.py:21]: (1) that line is too long

File: src/evaluators/gt_ss.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            sp_raw[P] = p
+        sp_mean  = sp_raw.groupby(by=[NETWORK, PROTOCOL, ACTOR, P]).mean().reset_index()
+        sp_mean = sp_mean[(sp_mean[PROTOCOL] == protocol) & (sp_mean[P] == p)]
+        sp_mean = sp_mean.sort_values(
+            [EXPOSED, SIMULATION_LENGTH, PEAK_INFECTED, PEAK_ITERATION],
+            ascending=[False, True, True, False]
+        )
+        return sp_mean.iloc[:budget][ACTOR].tolist()
+
+    def __call__(self, net_type: str, net_name: str, protocol: Literal["OR", "AND"], p: float, **kwargs):
```
Comment (mateuszStolarski) [inline src/evaluators/gt_ss.py:47]: (1) too long line

File: _configs/eval_ssm.yaml (hunk 1 above)
Comment (anty-filidor) [reply _configs/eval_ssm.yaml:8]: I'm sorry, but that would take too much time

File: src/evaluators/gt_ss.py (hunk 2 above)
Comment (anty-filidor) [reply src/evaluators/gt_ss.py:21]: applied

File: src/evaluators/gt_ss.py (hunk 3 above)
Comment (anty-filidor) [reply src/evaluators/gt_ss.py:47]: remark applied

**Review by mateuszStolarski — APPROVED** [review]

**Review by adampirog — APPROVED** [review]

File: src/evaluators/loader.py [hunk 4]
```diff
@@ -14,20 +17,38 @@ class SeedSet:
...
-def load_infmax_model(config: dict[str, Any], random_seed: int, nb_seeds: int) -> Any:
-    if config["name"] in {"MultiNode2VecKMeans", "MultiNode2VecKMeansAuto"}:
+def load_infmax_model(
+    config_infmax: dict[str, Any],
+    config_icm: dict[str, Any],
+    random_seed: int,
+    nb_seeds: int,
+) -> Any:
+    if config_infmax["class"] in {"MultiNode2VecKMeans", "MultiNode2VecKMeansAuto"}:
         from multi_node2vec_kmeans.loader import load_model
```
[the remark this answers is not in the document]
Comment (anty-filidor) [reply src/evaluators/loader.py:39]: that's intended to make this code working even if mn2v_km is not installed

File: src/evaluators/gt_ss.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        )
+        return sp_mean.iloc[:budget][ACTOR].tolist()
+
+    def __call__(
+            self,
+            net_type: str,
+            net_name: str,
+            protocol: Literal["OR", "AND"],
+            p: float,
+            **kwargs,
```
[the remark this answers is not in the document]
Comment (anty-filidor) [reply src/evaluators/gt_ss.py:58]: that's to unify interfaces of all infmax methods

**Pull request merged**